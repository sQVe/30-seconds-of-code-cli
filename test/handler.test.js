global.Math.random = () => 0.6

const snippets = [
  { id: 'bar', tags: ['array'] },
  { id: 'foobar', tags: ['array'] },
  { id: 'foo', tags: ['array'] },
  { id: 'fefoolo', tags: ['object'] },
  { id: 'bazfoo', tags: ['object'] },
]

import {
  getSnippet,
  getSnippetsByTag,
  randomSnippet,
  searchSnippets,
} from '../src/handler'

describe('Handler', () => {
  describe('getSnippet', () => {
    it('should find snippet by id', () => {
      expect(getSnippet(snippets, 'foo')).toEqual(snippets[2])
    })
  })

  describe('getSnippetsByTag', () => {
    it('should find snippets by tag', () => {
      expect(getSnippetsByTag(snippets, 'array')).toEqual(snippets.slice(0, 3))
    })
  })

  describe('randomSnippet', () => {
    it('should get random snippet', () => {
      expect(randomSnippet(snippets)).toEqual(snippets[3])
    })
  })

  describe('searchSnippets', () => {
    it('should find snippets fuzzy matching query', () => {
      expect(searchSnippets(snippets, 'foo')).toMatchSnapshot()
    })

    it('should sort by relevance', () => {
      const [x, y, z] = searchSnippets(snippets, 'foo')

      expect(x.relevance >= y.relevance && y.relevance >= z.relevance).toBe(
        true
      )
    })
  })
})
