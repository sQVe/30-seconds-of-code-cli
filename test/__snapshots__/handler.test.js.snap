// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handler searchSnippets should find snippets fuzzy matching query 1`] = `
Array [
  Object {
    "id": "foo",
    "relevance": 30,
    "tags": Array [
      "array",
    ],
  },
  Object {
    "id": "foobar",
    "relevance": 5.625,
    "tags": Array [
      "array",
    ],
  },
  Object {
    "id": "bazfoo",
    "relevance": 5.625,
    "tags": Array [
      "object",
    ],
  },
  Object {
    "id": "fefoolo",
    "relevance": 3,
    "tags": Array [
      "object",
    ],
  },
]
`;
